### PR TITLE
Block expired patients from being transfered

### LINKED ISSUE #6022: Marking Death on patient profile
**Is your feature request related to a problem? Please describe.**
In CARE an expired patient can be re-admitted in another encounter. This must not be allowed. A patient, once marked dead, the record must not be editable.

**Describe the solution you'd like**

- [x] Once the patient is discharged marked as expired, the patient file must not be transferred to another hospital/ must not be able to have another encounter on top of it. 
In the patient transfer form, the dropdown to select patient, an expired patients name should appear disabled (in grey) with "(expired)"added against the name
Screenshot 2024-07-08 at 10 29 24 AM

- [x] show a message stating that expired patient data cannot be edited later. In discharge pop-up, when Expired is selected in reason for discharge, change the text "Caution: this action is irreversible" to "Caution: Once a patient is marked as expired, the patient file cannot be transferred or edited. Please proceed with caution". Make this text appear in red colour. 

Screenshot 2024-07-08 at 10 33 56 AM

@gigincg Please comment

### PR DESCRIPTION
**Pull request by khavinshankar**
## Proposed Changes

- Fixes #6022 
- disabled expired patients in the patient transfer form
- altered warning text and color in discharge patient modal
- Depends on https://github.com/coronasafe/care/pull/2293

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

File: src/Components/Facility/DischargeModal.tsx [hunk 1]
```diff
@@ -210,9 +210,12 @@ const DischargeModal = ({
...
         <div>
           <p>Discharge patient from CARE</p>
-          <span className="mt-1 flex gap-1 text-sm font-medium text-warning-500">
+          <span className="mt-1 flex gap-1 text-sm font-medium text-danger-500">
             <CareIcon icon="l-exclamation-triangle" className="text-base" />
-            <p>Caution: this action is irreversible.</p>
+            <p>
+              Caution: Once a patient is marked as expired, the patient file
+              cannot be transferred or edited. Please proceed with caution.
+            </p>
```
Comment (rithviknishad) [inline src/Components/Facility/DischargeModal.tsx]: This caution can be shown when the selected reason is expired right? The original caution notice ("this action is irreversible") still holds true for non-expired situations too right.

**Review by rithviknishad — APPROVED** [review]

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
@khavinshankar 
image
image

in the patient transfer form, dropdown, as per issue, it should mention expired, and greyed out

**Comment by khavinshankar** [comment]
@nihal467 added expired near the option
image

Please check.

**Review by nihal467 — APPROVED** [review]

**Comment by nihal467** [comment]
@gigincg can you review and merge it

**Review by gigincg — APPROVED** [review]

**Pull request merged**